bindings/wasm/src/account/wasm_account/account_builder.rs

```diff
@@ -0,0 +1,171 @@
+/// used to store identities, and the same {@link Client} used to publish identities to the Tangle.
+///
+/// The configuration on the other hand is cloned, and therefore unique for each built account.
+/// This means a builder can be reconfigured in-between account creations, without affecting
+/// the configuration of previously built accounts.
+#[wasm_bindgen(js_name = AccountBuilder)]
+pub struct WasmAccountBuilder(Rc<RefCell<AccountBuilderRc>>);
+
+#[wasm_bindgen(js_class = AccountBuilder)]
+impl WasmAccountBuilder {
+  /// Creates a new `AccountBuilder`.
+  #[wasm_bindgen(constructor)]
+  pub fn new(options: Option<AccountBuilderOptions>) -> Result<WasmAccountBuilder> {
+    let mut builder: AccountBuilderRc = AccountBuilderRc::new();
+
+    if let Some(builder_options) = options {
+      if let Some(autopublish) = builder_options.autopublish() {
+        builder = builder.autopublish(autopublish);
+      }
+
+      if let Some(autosave) = builder_options.autosave().into_serde().wasm_result()? {
+        builder = builder.autosave(autosave);
+      }
+
+      if let Some(config) = builder_options.clientConfig() {
+        let client = executor::block_on(ClientBuilder::try_from(config)?.build()).wasm_result()?;
+        builder = builder.client(Rc::new(client));
+      };
+
+      if let Some(storage) = builder_options.storage() {
+        builder = builder.storage(AccountStorage::Custom(Arc::new(storage)));
+      }
+    }
+
+    Ok(Self(Rc::new(RefCell::new(builder))))
+  }
+
+  /// Loads an existing identity with the specified `did` using the current builder configuration.
+  /// The identity must exist in the configured `Storage`.
+  #[wasm_bindgen(js_name = loadIdentity)]
+  pub fn load_identity(&mut self, did: &WasmDID) -> Result<PromiseAccount> {
+    let builder: Rc<RefCell<AccountBuilderRc>> = self.0.clone();
+    let did: IotaDID = did.0.clone();
+    let promise: Promise = future_to_promise(async move {
+      builder
+        .as_ref()
+        .borrow_mut()
+        .load_identity(did)
+        .await
+        .map(WasmAccount::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+    Ok(promise.unchecked_into::<PromiseAccount>())
+  }
+
+  /// Creates a new identity based on the builder configuration and returns
+  /// an {@link Account} object to manage it.
+  ///
+  /// The identity is stored locally in the `Storage`. The DID network is automatically determined
+  /// by the {@link Client} used to publish it.
+  ///
+  /// @See {@link IdentitySetup} to customize the identity creation.
+  #[wasm_bindgen(js_name = createIdentity)]
+  pub fn create_identity(&mut self, identity_setup: Option<WasmIdentitySetup>) -> Result<PromiseAccount> {
+    let setup: IdentitySetup = {
+      match identity_setup {
+        None => IdentitySetup::default(),
+        Some(wasm_identity_setup) => IdentitySetup::try_from(wasm_identity_setup)?,
+      }
+    };
```

Comment:
Nit: can be condensed.

```suggestion
    let setup: IdentitySetup = identity_setup.map(IdentitySetup::try_from).transpose()?.unwrap_or_default();
```